packages/discussions/src/posts.ts

```diff
@@ -0,0 +1,193 @@
+import { request } from "./request";
+import {
+  IQueryPostsOptions,
+  IQueryChannelPostsOptions,
+  ICreatePostOptions,
+  ICreateChannelPostOptions,
+  ICreateReplyOptions,
+  ICreateChannelReplyOptions,
+  IGetPostOptions,
+  IGetChannelPostOptions,
+  IDeletePostOptions,
+  IDeleteChannelPostOptions,
+  IUpdatePostOptions,
+  IUpdateChannelPostOptions,
+  IUpdatePostSharingOptions,
+  IUpdateChannelPostSharingOptions,
+  IUpdatePostStatusOptions,
+  IUpdateChannelPostStatusOptions,
+  INestPagination,
+  IPostDTO,
+  INestDeleteResult
+} from "./types";
+
+/**
+ * search posts
+ *
+ * @export
+ * @param {(IQueryPostsOptions | IQueryChannelPostsOptions)} options
+ * @return {*}  {Promise<INestPagination<IPostDTO>>}
+ */
+export function searchPosts(
+  options: IQueryPostsOptions | IQueryChannelPostsOptions
+): Promise<INestPagination<IPostDTO>> {
+  let url = `/posts`;
+  if (options.params.hasOwnProperty("channelId")) {
+    const {
+      params: { channelId }
+    } = options as IQueryChannelPostsOptions;
+    url = `/channels/${channelId}` + url;
```

Comment:
Why just one method? Why not a method `searchPost` and `searchChannel`? It allows code to be more explicit and you remove the `if` condition and you force validation by parameters.

Reply:
So this method is searching posts; `IQueryChannelPostsOptions` scopes the search to a single channel, while `IQueryPostsOptions` can search many channels. Does that change your opinion?

Otherwise, I thought having two methods, `searchPosts` and `searchChannelPosts`, to be redundant and consolidated to a single method for the convenience of dev users.